### PR TITLE
feat(core): expose UnknownActorRef

### PR DESCRIPTION
**Pull request by boneskull**
`AnyActorRef` is problematic for some use cases, because the result of the `AnyActorRef['getSnapshot']` function is of type `any`.

However, changing `AnyActorRef` to use a narrow type in `ActorRef`'s first type argument, `TSnapshot`, breaks conditional types which perform inference based on `TSnapshot`.

This change introduces `UnknownActorRef`, which is like `AnyActorRef`, but is not intended to be inferred from in future conditional types. A consumer can use `UnknownActorRef` like so:

```ts
const actor: UnknownActorRef = getSomeActor();

// inferred as `Snapshot<unknown>`
const snapshot = actor.getSnapshot();

// do things w/ snapshot
```

`AnyActorRef` would return `any` from `actor.getSnapshot()`, which make it unsuitable for this use-case in a strictly-typed environment.

### CODE REVIEW

**Comment by boneskull** [comment]
regardless; if this (or the other suggestion) makes sense, I'll go in and make the change and fix the type tests.

**Comment by davidkpiano** [comment]
Thank you for this! It seems like TS is complaining in some of the test files (you can run `yarn typecheck` to check) 🕵️ 

cc. @Andarist

**Comment by boneskull** [comment]
@davidkpiano I don't quite understand why the 3rd `any` in `AnyActorRef` was removed in dbeafeb25eed63ee1d2b027f10bc63d9937ab073 (#4905).  That change seems to break some stuff anyhow.  I'll create a simple reproduction.

**Comment by boneskull** [comment]
See https://github.com/statelyai/xstate/issues/4931 for the "other problem" I mentioned above.

**Comment by davidkpiano** [comment]
There may be some overlap with #4932

**Comment by boneskull** [comment]
@davidkpiano Yeah.  I'll wait until #4932 is merged then retry on top of it.

**Comment by davidkpiano** [comment]
@boneskull It's merged; please merge `main` and try again?

**Comment by boneskull** [comment]
looking

**Comment by boneskull** [comment]
@davidkpiano OK, I rebased, but changed my strategy. Please read updated issue description

File: packages/core/src/types.ts
Comment (Andarist) [inline packages/core/src/types.ts:2189]: I'd call this `UnknownActorRef`. It might fit better some other existing types of this kind

Comment (boneskull) [reply packages/core/src/types.ts:2189]: works for me

**Comment by boneskull** [comment]
@davidkpiano I've added a changeset for this.  I'm calling it "patch" since it doesn't affect any functionality. Happy to change it

**Review by Andarist — APPROVED** [review]

**Pull request merged**
